front/pages/api/w/[wId]/vaults/[vId]/data_source_views/index.ts

```diff
@@ -50,15 +67,72 @@ async function handler(
           ? req.query.category
           : null;
 
-      const dataSourceViews = await DataSourceViewResource.listByVault(
-        auth,
-        vault
-      );
+      const dataSourceViews = (
+        await DataSourceViewResource.listByVault(auth, vault)
+      )
+        .map((ds) => ds.toJSON())
+        .filter((d) => !category || d.category === category);
+
+      if (req.query.includeConnectorDetails) {
+        const enhancedDataSourceViews = await Promise.all(
+          dataSourceViews.map(async (dataSourceView) => {
+            const dataSource = dataSourceView.dataSource;
+            if (!isManaged(dataSource)) {
```

Comment:
IMO, `isManaged` should be a proper typeguard that narrows down the type so `connectorProvider` is always defined.